sizeOptimizer.cpp

```diff
@@ -37,11 +37,15 @@ std::vector<const Transform *> *SizeOptimizer::optimize(size_t nBest,
                                                         int maxPercIncrease,
                                                         int maxMemMB,
                                                         bool squareOnly,
-                                                        bool crop) {
+                                                        bool crop, int rank) {
   std::vector<GeneralTransform> preoptimized;
   for (auto in : input) {
-    std::vector<GeneralTransform> *tmp =
-        optimizeXYZ(in, nBest, maxPercIncrease, squareOnly, crop);
+    std::vector<GeneralTransform> *tmp;
+    if(rank == 3){
+      tmp = optimizeXYZ_3D(in, nBest, maxPercIncrease, squareOnly, crop);
```

Comment:
no need to decide here. Check it within the optimizeXYZ (and call appropriate version there)